lib/aliyun/oss/bucket.rb

```diff
@@ -47,6 +47,38 @@ def logging=(logging)
         end
       end
 
+      # 获取Bucket的versioning配置
+      # @return [BucketVersioning] Bucket的versioning配置
+      def versioning
+        @protocol.get_bucket_versioning(name)
+      end
+
+      # 设置Bucket的versioning配置
+      # @param versioning [BucketVersioning] versioning配置
+      def versioning=(versioning)
+        if versioning.enabled?
+          @protocol.put_bucket_versioning(name, versioning)
+        else
+          @protocol.delete_bucket_versioning(name)
```

Comment:
This delete_bucket_versioning is not required because this function is not supported by server.

Reply:
and the BuckerVersioning will be changed below:

    # Attributes:
    # * status [String] the bucket versioning status, can be 'Enabled' and 'Suspended'
    # @example Enable bucket versioning
    #  bucket.versioning = BucketVersioning.new(:status => 'Enabled')
    # @example Suspend bucket versioning
    #  bucket.versioning = BucketVersioning.new(:status => 'Suspended')
    class BucketVersioning < Common::Struct::Base
      attrs :status
    end